File: Chapter04/quickSort.py
from array import array

def quickSort(array:list):
    if(len(array)<2):
        return array
    else:
        pivot=array[0]
        #小于基准值的区间
        less=[]
        #大于基准值的区间
        greater=[]
        for i in array[1:]:
            if(i<=pivot):
                less.append(i)
            else:
                greater.append(i)
        return quickSort(less)+[pivot]+quickSort(greater)        
array=[10,5,2,3]
print(quickSort(array))        

File: Chapter04/quickSort.js
const quickSort=array=>{
    if(array.length<2){
        return array;
    }

    let pivot=array[0],//基准值
        less=[],//小于基准值的区间
        greater=[];//大于基准值的区间
    for (const iterator of array.slice(1)) {
        if(pivot>=iterator)
            less.push(iterator)
        else
            greater.push(iterator)    
    }        
    return [...quickSort(less),pivot,...quickSort(greater)];
}

let array=[10,5,2,3];
console.log(quickSort(array));